### PR TITLE
Add preliminary documentation of suffix based specification of units

### LINKED ISSUE #3345: Add documentation for unit specification of input parameter values
There is some automatic conversion that can happen dmstor_ctx() (in `src/dmstor.cpp`) such that the input values for some parameters can be interpreted as either decimal degrees or radians, and converted appropriately into radians (i.e. converted into radians for the value that is returned in PROJVALUE.f).

This feature request adds some preliminary documentation for this suffix-based unit specification and the convention that decimal degrees is the default interpretation.

### PR DESCRIPTION
**Pull request by sval-dev**
Add preliminary documentation of suffix based specification of units for input parameter values.

 - [X] Closes #3345
 - [X] Added clear title that can be used to generate release notes

### CODE REVIEW

File: docs/source/usage/projections.rst [hunk 1]
```diff
@@ -64,6 +64,32 @@ Scaling of output units can be done by applying the ``+k_0`` argument. The
...
+    | °              |                     |
+    +----------------+---------------------+
+    | r              | Radians             |
+    +----------------+                     +
+    | R              |                     |
+    +----------------+---------------------+
+
+Example of use.  The longitude ``long=90``, can also be expressed more explictly
+with units of decimal degrees as ``long=90d`` and can also be expressed in radian
+units as ``long=1.5708r`` (approximately).
```
Comment (kbevers) [inline docs/source/usage/projections.rst]: Reworded to use a well-known projection parameter:

```suggestion
Example of use.  The longitude of the central meridian ``+lon_0=90``, can also be expressed more explictly
with units of decimal degrees as ``+lon_0=90d`` or in radian
units as ``+lon_0=1.5708r`` (approximately).
```

**Review by kbevers — CHANGES_REQUESTED** [review]
Thanks for adding this! I've suggested a different wording of the example using the central meridian.

It would be great to mention this feature in the docs for each of the generic options where it is useful. Perhaps just a simple example of how to give the value as radian and a link back to the new section of the docs provided in this PR. I believe the relevant options are `+lon_0`, `+lat_0`, `+lat_1`, `lat_2` and `+lat_ts`. They described here: https://github.com/OSGeo/PROJ/tree/master/docs/source/operations/options

**Review by kbevers — APPROVED** [review]
This looks great, thanks for brushing the docs up a bit

**Comment by sval-dev** [comment]
Added some examples to the suggestion operation options.

**Comment by sval-dev** [comment]
> This looks great, thanks for brushing the docs up a bit

Glad to help!

**Pull request merged**